### PR TITLE
[DAT-18790] added defaultValue snapshotting

### PR DESCRIPTION
**Pull request by KushnirykOleh**

### CODE REVIEW

**Comment by SvampX** [comment]
Should work for most cases except:
Default value is calculated by a function with multiple arguments like `eventDate DATE DEFAULT coalesce(CAST(CURRENT_TIMESTAMP() AS DATE), CAST('2024.09.09' AS DATE))`

**Review by filipelautert — APPROVED** [review]

File: src/main/java/liquibase/ext/databricks/snapshot/jvm/ColumnSnapshotGeneratorDatabricks.java [hunk 1]
```diff
@@ -43,4 +59,71 @@ protected DataType readDataType(CachedRow columnMetadataResultSet, Column column
...
     }
+
+    @Override
+    protected DatabaseObject snapshotObject(DatabaseObject example, DatabaseSnapshot snapshot) throws DatabaseException {
+        //This should work after fix on Databricks side
+        if (example instanceof Column) {
+            Column column = (Column) super.snapshotObject(example, snapshot);
+            //These two are used too often, avoiding them? otherwise there would be too much DB calls
+            if (!column.getRelation().getName().equalsIgnoreCase("databasechangelog")
+                    && !column.getRelation().getName().equalsIgnoreCase("databasechangeloglock")) {
```
Comment (KushnirykOleh) [inline src/main/java/liquibase/ext/databricks/snapshot/jvm/ColumnSnapshotGeneratorDatabricks.java:70]: Even with skipping DBCL and DBCLL we are making additional call for every column. We have info for all columns in `show create table`.
column.getRelation() has all the columns, so we can iterate over them and add defaultValue information for all columns at once

File: src/main/java/liquibase/ext/databricks/snapshot/jvm/ColumnSnapshotGeneratorDatabricks.java [hunk 2]
```diff
@@ -43,4 +59,71 @@ protected DataType readDataType(CachedRow columnMetadataResultSet, Column column
...
         return super.readDataType(columnMetadataResultSet, column, database);
     }
+
+    @Override
+    protected DatabaseObject snapshotObject(DatabaseObject example, DatabaseSnapshot snapshot) throws DatabaseException {
+        //This should work after fix on Databricks side
+        if (example instanceof Column) {
+            Column column = (Column) super.snapshotObject(example, snapshot);
+            //These two are used too often, avoiding them? otherwise there would be too much DB calls
+            if (!column.getRelation().getName().equalsIgnoreCase("databasechangelog")
```
Comment (KushnirykOleh) [inline src/main/java/liquibase/ext/databricks/snapshot/jvm/ColumnSnapshotGeneratorDatabricks.java:69]: tracking table can have custom names. 
better to use `database.getDatabaseChangeLogTableName()` and  `database.getDatabaseChangeLogLockTableName()`

File: src/main/java/liquibase/ext/databricks/snapshot/jvm/ColumnSnapshotGeneratorDatabricks.java (hunk 1 above)
Comment (filipelautert) [reply src/main/java/liquibase/ext/databricks/snapshot/jvm/ColumnSnapshotGeneratorDatabricks.java:70]: @SvampX as discussed, maybe DatabaseSnapshot.snapshotScratchPad could be useful to cache this information between executions.

**Review by filipelautert — CHANGES_REQUESTED** [review]

File: src/main/java/liquibase/ext/databricks/snapshot/jvm/ColumnSnapshotGeneratorDatabricks.java (hunk 1 above)
Comment (SvampX) [reply src/main/java/liquibase/ext/databricks/snapshot/jvm/ColumnSnapshotGeneratorDatabricks.java]: Moved show create table call to the Table snapshot generator. Storing create table statement in the `snapshotScratchPad` suggested by @filipelautert  with a show create table query as key. I believe such keys should be unique within a database as the query has full table name with catalog and schema.

**Review by filipelautert — APPROVED** [review]
Nice!

**Review by KushnirykOleh — COMMENTED** [review]
Looks good.
(I can't approve as i opened PR originally)

**Pull request merged**